spvc/src/main.cc

```diff
@@ -0,0 +1,179 @@
+      return true;
+    }
+    if (++(*index) >= argc) return false;
+    *option_argument = argv[*index];
+    return true;
+  }
+}
+
+const char kBuildVersion[] = ""
+    // TODO #include "build-version.inc"
+    ;
+
+bool ReadFile(const std::string& path, std::vector<uint32_t>* out) {
+  FILE* file =
+      path == "-" ? freopen(nullptr, "rb", stdin) : fopen(path.c_str(), "rb");
+  if (!file) {
+    std::cerr << "Failed to open SPIR-V file: " << path << std::endl;
+    return false;
+  }
+
+  fseek(file, 0, SEEK_END);
+  out->resize(ftell(file) / sizeof((*out)[0]));
+  rewind(file);
+
+  if (fread(out->data(), sizeof((*out)[0]), out->size(), file) != out->size()) {
+    std::cerr << "Failed to read SPIR-V file: " << path << std::endl;
+    out->clear();
+    return false;
+  }
+
+  fclose(file);
+  return true;
+}
+
+}  // anonymous namespace
+
+int main(int argc, char** argv) {
+  shaderc_spvc::Compiler compiler;
+  shaderc_spvc::CompileOptions options;
+  std::vector<uint32_t> input;
+  string_piece output_path;
+
+  for (int i = 1; i < argc; ++i) {
+    const string_piece arg = argv[i];
+    if (arg == "--help") {
+      ::PrintHelp(&std::cout);
+      return 0;
+    } else if (arg == "--version") {
+      std::cout << kBuildVersion << std::endl;
+      std::cout << "Target: " << spvTargetEnvDescription(SPV_ENV_UNIVERSAL_1_0)
+                << std::endl;
+      return 0;
+    } else if (arg.starts_with("-o")) {
+      if (!GetOptionArgument(argc, argv, &i, "-o", &output_path)) {
+        std::cerr
+            << "spvc: error: argument to '-o' is missing (expected 1 value)"
+            << std::endl;
+        return 1;
+      }
+    } else if (arg == "--remove-unused-variables") {
+      // TODO
+    } else if (arg.starts_with("--validate=")) {
+      string_piece target_env;
+      GetOptionArgument(argc, argv, &i, "--validate=", &target_env);
+      if (target_env == "vulkan") {
+        options.SetTargetEnvironment(shaderc_target_env_vulkan,
+                                     shaderc_env_version_vulkan_1_0);
+      } else if (target_env == "vulkan1.0") {
+        options.SetTargetEnvironment(shaderc_target_env_vulkan,
+                                     shaderc_env_version_vulkan_1_0);
+      } else if (target_env == "vulkan1.1") {
+        options.SetTargetEnvironment(shaderc_target_env_vulkan,
+                                     shaderc_env_version_vulkan_1_1);
+      } else if (target_env == "opengl") {
+        options.SetTargetEnvironment(shaderc_target_env_opengl,
+                                     shaderc_env_version_vulkan_1_1);
+      } else if (target_env == "opengl4.5") {
+        options.SetTargetEnvironment(shaderc_target_env_opengl,
+                                     shaderc_env_version_vulkan_1_1);
+      } else {
+        std::cerr << "spvc: error: invalid value '" << target_env
+                  << "' in '--validate=" << target_env << "'" << std::endl;
+        return 1;
+      }
+    } else {
```

Comment:
This can just be an else if

Reply:
I think I like it better as is because the if else chain deals only with 'arg' and the inside code does what it needs to do for each specific arg.
You woudn't want "if (arg.starts_with("-o") && GetOptionArgument(....))" would you?